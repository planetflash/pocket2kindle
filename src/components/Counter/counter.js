import React from "react";

import Button from "components/Button";

class Counter extends React.Component {
  state = {
    count: 0
  };

  increment = () => {
    this.setState(prevState => {
      return {
        count: prevState.count + 1
      };
    });
  };

  render() {
    return (
      <div>
        <p>Current count: {this.state.count}</p>
        <Button onClick={this.increment} text={"Increment Count"} />
      </div>
    );
  }
}

export default Counter;
