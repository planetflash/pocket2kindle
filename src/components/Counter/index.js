import React from "react";

import Button from "components/Button";

class Counter extends React.Component {
  state = {
    count: 0
  };

  makeIncrementer = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount
    }));

  increment = this.makeIncrementer(1);
  decrement = this.makeIncrementer(-1);

  render() {
    return (
      <div>
        <p>Current count: {this.state.count}</p>
        <Button
          className="increment"
          onClick={this.increment}
          text={"Increment Count"}
        />
        <Button
          className="decrement"
          onClick={this.decrement}
          text={"Decrement Count"}
        />
      </div>
    );
  }
}

export default Counter;
