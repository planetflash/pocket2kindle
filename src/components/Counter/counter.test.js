import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Counter from "./";

describe("Counter component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Counter />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("starts witn a count of 0", () => {
    const wrapper = shallow(<Counter />);
    const text = wrapper.find("p").text();
    expect(text).toEqual("Current count: 0");
  });

  it("can increment the count when the button is clicked", () => {
    const wrapper = shallow(<Counter />);
    const incrementBtn = wrapper.find("Button.increment");
    incrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Current count: 1");
  });

  it("can decrement the count when the button is clicked", () => {
    const wrapper = shallow(<Counter />);
    const decrementBtn = wrapper.find("Button.decrement");
    decrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Current count: -1");
  });
});
