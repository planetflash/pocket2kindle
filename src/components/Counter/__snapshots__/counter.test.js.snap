// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the snapshot 1`] = `
<div>
  <p>
    Current count: 
    0
  </p>
  <div>
    <a
      className="src-components-Button-___style__btn___2qMxX"
      onClick={[Function]}
    >
      <span>
        +++
      </span>
    </a>
  </div>
  <div>
    <a
      className="src-components-Button-___style__btn___2qMxX"
      onClick={[Function]}
    >
      <span>
        ---
      </span>
    </a>
  </div>
</div>
`;
